.github/workflows/kind-e2e-upgrade.yaml

```diff
@@ -66,3 +63,11 @@ jobs:
     - name: Run upgrade tests
       run: |
         ./test/e2e-upgrade-kind.sh
+
+    # Collect debug logs on failure
+    - uses: chainguard-dev/actions/kind-diag@main
+      if: ${{ failure() }}
+      with:
+        cluster-resources: nodes,namespaces,crds
+        namespace-resources: configmaps,pods,svc,king
+        artifact-name: logs-${{ matrix.k8s-version}}-${{ matrix.upstream-traffic }}
```

Comment:
While I've never actually used it myself, I'm LGTM based on the following points:
- It's also used in other repositories.
  - https://github.com/knative-extensions/net-contour/blob/main/.github/workflows/kind-e2e.yaml#L47
- From a quick glance, it seems to only handle read operations.
  - https://github.com/chainguard-dev/actions/blob/main/kind-diag/action.yaml
- Even if this step crashes due to errors, it shouldn't cause any major issues.